pkg/api/v1beta3/types.go

```diff
@@ -170,12 +170,14 @@ type Volume struct {
 // VolumeSource represents the source location of a valume to mount.
 // Only one of its members may be specified.
 type VolumeSource struct {
-	// HostDir represents a pre-existing directory on the host machine that is directly
-	// exposed to the container. This is generally used for system agents or other privileged
-	// things that are allowed to see the host machine. Most containers will NOT need this.
+	// HostPath represents a pre-existing file or directory on the host
+	// machine that is directly exposed to the container. This is generally
+	// used for system agents or other privileged things that are allowed
+	// to see the host machine. Most containers will NOT need this.
 	// TODO(jonesdl) We need to restrict who can use host directory mounts and who can/can not
 	// mount host directories as read/write.
-	HostDir *HostDir `json:"hostDir"`
+	HostPath *HostPath `json:"hostPath"`
+	HostDir  *HostPath `json:"hostDir"` // DEPRECATED: use HostPath instead
```

Comment:
The main Volumes[] structure has not changed at all between v1beta1 and
v1beta3.  I thought it was fair to allow the old name.  we don't have to,
but I expect a bit of pain

On Tue, Jan 20, 2015 at 3:02 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1beta3/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3517#discussion_r23264713>
> :
>
> >  	// TODO(jonesdl) We need to restrict who can use host directory mounts and who can/can not
> >  	// mount host directories as read/write.
> > -	HostDir *HostDir `json:"hostDir"`
> > +	HostPath *HostPath `json:"hostPath"`
> > +	HostDir  *HostPath `json:"hostDir"` // DEPRECATED: use HostPath instead
>
> Could we not start v1beta3 with deprecated fields? Why is this needed?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3517/files#r23264713>
> .
>